src/helpers/accessiblity.ts

```diff
@@ -0,0 +1,52 @@
+import { StyleSheet } from 'react-native';
+import { ReactTestInstance } from 'react-test-renderer';
+
+export function isInaccessible(instance: ReactTestInstance | null): boolean {
+  if (!instance) {
+    return true;
+  }
+
+  // Android: importantForAccessibility
+  // See: https://reactnative.dev/docs/accessibility#importantforaccessibility-android
+  if (instance.props.importantForAccessibility === 'no') return true;
+
+  let current: ReactTestInstance | null = instance;
+  while (current) {
+    if (isSubtreeInaccessible(current)) {
+      return true;
+    }
+
+    current = current.parent;
+  }
+
+  return false;
+}
+
+function isSubtreeInaccessible(
+  instance: ReactTestInstance | null | undefined
+): boolean {
+  if (!instance) {
+    return true;
+  }
+
+  // TODO implement iOS: accessibilityViewIsModal
+  // The hard part is to implement this to look only for host views
+  // See: https://reactnative.dev/docs/accessibility#accessibilityviewismodal-ios
+  //   if (instance.parent?.children.some((child) => child.accessibilityViewIsModal))
+  //     return true;
+
+  // iOS: accessibilityElementsHidden
+  // See: https://reactnative.dev/docs/accessibility#accessibilityelementshidden-ios
+  if (instance.props.accessibilityElementsHidden) return true;
+
+  // Android: importantForAccessibility
+  // See: https://reactnative.dev/docs/accessibility#importantforaccessibility-android
+  if (instance.props.importantForAccessibility === 'no-hide-descendants')
+    return true;
+
+  const flatStyle = StyleSheet.flatten(instance.props.style) ?? {};
+  if (flatStyle.display === 'none') return true;
+  if (flatStyle.opacity === 0) return true;
```

Comment:
Are zero-opacity items inaccessible? Would an item with opacity 0 would be available for screen readers?

Reply:
I had not found any clear statement in docs that supports, or rejects, being read by VoiceOver screen reader. On the web it is reportedly not ignored by a screen reader, so I will need to conduct an actual experiment to check that.

#### DTL comparison

Seems like the DTL implementation does not check for `opacity: 0`, and uses only the following:
* `element.hidden === true` (incl. subtree)
* `element.getAttribute('aria-hidden') === 'true'` (incl. subtree)
* `window.getComputedStyle(element).display === 'none'` (incl. subtree)
* `window.getComputedStyle(element).visibility === 'hidden'` (only directly on element)

See: https://github.com/testing-library/dom-testing-library/blob/0226aea74e873ba96dae414edc533c33b1e51867/src/role-helpers.js#L15

Also `hidden` does not seem to be affected by `opacity` style.

Reply:
Let's remove the opacity check then :)

Reply:
not at the moment. Let's watch this space